pkg/chains/storage/tekton/tekton.go

```diff
@@ -110,18 +111,37 @@ func (b *Backend) retrieveAnnotationValue(annotationKey string, decode bool) (st
 }
 
 // RetrieveSignature retrieve the signature stored in the taskrun.
-func (b *Backend) RetrieveSignature(opts config.StorageOpts) (string, error) {
+func (b *Backend) RetrieveSignatures(opts config.StorageOpts) (map[string][]string, error) {
 	b.logger.Infof("Retrieving signature on TaskRun %s/%s", b.tr.Namespace, b.tr.Name)
-	return b.retrieveAnnotationValue(fmt.Sprintf(SignatureAnnotationFormat, opts.Key), true)
+	signature, err := b.retrieveAnnotationValue(fmt.Sprintf(SignatureAnnotationFormat, opts.Key), true)
+	if err != nil {
+		return nil, err
+	}
+
+	m := make(map[string][]string)
+	for _, res := range b.tr.Status.TaskRunResults {
+		if strings.HasSuffix(res.Name, "IMAGE_URL") {
+			m[res.Value] = []string{signature}
+			break
+		}
+	}
+	return m, nil
 }
 
 // RetrievePayload retrieve the payload stored in the taskrun.
-func (b *Backend) RetrievePayload(opts config.StorageOpts) (string, error) {
+func (b *Backend) RetrievePayloads(opts config.StorageOpts) (map[string]string, error) {
 	b.logger.Infof("Retrieving payload on TaskRun %s/%s", b.tr.Namespace, b.tr.Name)
 	payload, err := b.retrieveAnnotationValue(fmt.Sprintf(PayloadAnnotationFormat, opts.Key), true)
 	if err != nil {
-		return "", err
+		return nil, err
+	}
+	m := make(map[string]string)
+	for _, res := range b.tr.Status.TaskRunResults {
+		if strings.HasSuffix(res.Name, "IMAGE_URL") {
```

Comment:
Similarly, should we just be mapping `payloadAnnotation`:`payload` here?

Reply:
Addressed ✅ 

Like above, should we check there is only one image?